lib/fluent/supervisor.rb

```diff
@@ -243,82 +318,33 @@ def show_plugin_config
       exit 1
     end
 
-    def start_daemonize
-      @wait_daemonize_pipe_r, @wait_daemonize_pipe_w = IO.pipe
-
-      if fork
-        # console process
-        @wait_daemonize_pipe_w.close
-        @wait_daemonize_pipe_w = nil
-        wait_daemonize
-        exit 0
-      end
-
-      # daemonize intermediate process
-      @wait_daemonize_pipe_r.close
-      @wait_daemonize_pipe_r = nil
-
-      # in case the child process forked during run_configure
-      @wait_daemonize_pipe_w.fcntl(Fcntl::F_SETFD, Fcntl::FD_CLOEXEC)
-
-      Process.setsid
-      exit!(0) if fork
-      File.umask(0)
-
-      # supervisor process
-      @supervisor_pid = Process.pid
-    end
-
-    def wait_daemonize
-      supervisor_pid = @wait_daemonize_pipe_r.read
-      if supervisor_pid.empty?
-        # initialization failed
-        exit! 1
-      end
-
-      @wait_daemonize_pipe_r.close
-      @wait_daemonize_pipe_r = nil
-
-      # write pid file
-      File.open(@daemonize, "w") {|f|
-        f.write supervisor_pid
-      }
-    end
-
-    def finish_daemonize
-      if @wait_daemonize_pipe_w
-        STDIN.reopen("/dev/null")
-        STDOUT.reopen("/dev/null", "w")
-        STDERR.reopen("/dev/null", "w")
-        @wait_daemonize_pipe_w.write @supervisor_pid.to_s
-        @wait_daemonize_pipe_w.close
-        @wait_daemonize_pipe_w = nil
-      end
-    end
-
+    # TODO: move RPC Server in serverengine side
     def setup_rpc_server
       @rpc_server = RPC::Server.new(@rpc_endpoint, $log)
 
       # built-in RPC for signals
       @rpc_server.mount_proc('/api/processes.interruptWorkers') { |req, res|
         $log.debug "fluentd RPC got /api/processes.interruptWorkers request"
-        supervisor_sigint_handler
+        Process.kill :INT, $$
         nil
       }
       @rpc_server.mount_proc('/api/processes.killWorkers') { |req, res|
         $log.debug "fluentd RPC got /api/processes.killWorkers request"
-        supervisor_sigterm_handler
+        Process.kill :TERM, $$
         nil
       }
       @rpc_server.mount_proc('/api/plugins.flushBuffers') { |req, res|
         $log.debug "fluentd RPC got /api/plugins.flushBuffers request"
-        supervisor_sigusr1_handler
+        unless ServerEngine.windows?
+          Process.kill :USR1, $$
```

Comment:
If ServerEngine provides any supervisor-worker communication path, we should use it instead of signaling SIGUSR1 (because it's not supported by Windows, but we want to provide buffer force flushing even in Windows if possible).